Sources/MapboxDirections/RestStop.swift

```diff
@@ -22,4 +24,22 @@ public struct RestStop: Codable, Equatable {
     public init(type: StopType) {
         self.type = type
     }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        type = try container.decode(StopType.self, forKey: .type)
+        
+        try decodeForeignMembers(notKeyedBy: CodingKeys.self, with: decoder)
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(type, forKey: .type)
+        
+        try encodeForeignMembers(notKeyedBy: CodingKeys.self, to: encoder)
+    }
+    
+    public static func == (lhs: Self, rhs: Self) -> Bool {
+        return lhs.type == rhs.type
```

Comment:
In Turf, any struct that conforms to ForeignMemberContainer considers two instances with different foreign members to be unequal.

Reply:
But Turf basically provides a set of data containers. I am afraid that accounting for foreignMembers may introduce subtle bugs in Directions or Navigation logic when 'suddenly' objects will become unequal because of members which are unimportant for the SDK.

Reply:
Agreed, this is more likely for classes than structs anyhow.